File: 10-ScanCode/01-ScanCode(OC)/01-ScanCode(OC)/ViewController.m
//
//  ViewController.m
//  01-ScanCode(OC)
//
//  Created by 冯志浩 on 2017/4/10.
//  Copyright © 2017年 冯志浩. All rights reserved.
//
#define ScreenWidth [UIScreen mainScreen].bounds.size.width
#import "ViewController.h"
#import "ScanCodeViewController.h"
@interface ViewController ()
@property (nonatomic, strong) UIButton *pushButton;
@property (nonatomic, strong) UILabel *textLabel;
@property (nonatomic, strong) ScanCodeViewController *scanVC;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    _scanVC = [[ScanCodeViewController alloc]init];
    [self setupSubView];
}

- (void)setupSubView {
    _pushButton = [[UIButton alloc]initWithFrame:CGRectMake(100, 100, 200, 50)];
    [_pushButton setTitle:@"pushNextVC" forState:UIControlStateNormal];
    _pushButton.backgroundColor = [UIColor blackColor];
    [_pushButton addTarget:self action:@selector(pushButtonDidTouch) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:_pushButton];
    
    _textLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, 280, ScreenWidth, 50)];
    _textLabel.textColor = [UIColor blackColor];
    _textLabel.textAlignment = NSTextAlignmentCenter;
    _textLabel.text = @"text";
    [self.view addSubview:_textLabel];
}

- (void)pushButtonDidTouch {
    __weak typeof(self) weakSelf = self;
    _scanVC.resultBlock = ^(NSString *str){
        weakSelf.textLabel.text = str;
    };
    [self.navigationController pushViewController:_scanVC animated:YES];
}

@end

File: 08-ADScrollView/08-ADScrollView/ViewController.swift
//
//  ViewController.swift
//  08-ADScrollView
//
//  Created by 冯志浩 on 2016/11/28.
//  Copyright © 2016年 FZH. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    let SCREEN_WIDTH = UIScreen.main.bounds.size.width
    let SCREEN_HEIGHT = UIScreen.main.bounds.size.height
    override func viewDidLoad() {
        super.viewDidLoad()
        setupSubViews()
    }

    func setupSubViews() -> Void {
        let adScrollView = FZHADScrollView.init(frame: CGRect(x: 0, y: 100, width: SCREEN_WIDTH, height: 50))
        adScrollView.backgroundColor = UIColor.brown
        view.addSubview(adScrollView)
    }

}

